src/display/FunctionGraph/Parametric.tsx

```diff
@@ -18,33 +31,51 @@ const ParametricFunction: React.VFC<ParametricProps> = ({
   style = "solid",
   weight = 2,
   opacity = 1.0,
-  samples,
+  minimumSamplingDepth = 8,
   svgPathProps = {},
 }) => {
-  const { cssScale } = useScaleContext()
+  const { cssScale, scaleX, scaleY } = useScaleContext()
 
   const [tMin, tMax] = t
+  const areaThreshold = -0.1 / (scaleX(1) * scaleY(1))
+
+  const svgPath = useMemo(() => {
+    let pathDescriptor = "M "
+
+    function smartSmooth(
+      min: number,
+      max: number,
+      pushLeft: boolean,
+      pushRight: boolean,
+      depth = 0
+    ) {
+      const mid = (min + max) / 2
+
+      const xyMin = xy(min)
+      const xyMid = xy(mid)
+      const xyMax = xy(max)
 
-  // How many points to render
-  samples = samples || 1000
-  const dt = (tMax - tMin) / samples
+      const area = triangleArea(xyMin, xyMid, xyMax)
 
-  const points: [number, number][] = []
+      if (depth < minimumSamplingDepth || area > areaThreshold) {
+        smartSmooth(min, mid, true, false, depth + 1)
+        smartSmooth(mid, max, false, true, depth + 1)
+      } else {
+        if (pushLeft) pathDescriptor += `${xyMin[0]} ${xyMin[1]} L `
+        pathDescriptor += `${xyMid[0]} ${xyMid[1]} L `
+        if (pushRight) pathDescriptor += `${xyMax[0]} ${xyMax[1]} L `
+      }
+    }
 
-  const tAlmostMax = tMax - dt / 2
-  for (let t = tMin; dt > 0 ? t < tAlmostMax : t > tAlmostMax; t += dt) {
-    const [x, y] = xy(t)
-    points.push([x, y])
-  }
-  points.push(xy(tMax))
+    smartSmooth(tMin, tMax, true, true)
 
-  const pathDescriptor = "M " + points.map((p) => p.join(" ")).join(" L ")
+    return pathDescriptor.substring(0, pathDescriptor.length - 3)
+  }, [tMin, tMax, xy, areaThreshold])
 
   return (
     <path
-      d={pathDescriptor}
+      d={svgPath}
       strokeWidth={weight}
-      tabIndex={0}
```

Comment:
Fixes a bug where functions were focusable. This is from the initial commit. I think my reasoning is that keyboard users might want to focus on functions for some reason?